core/chains/evm/gas/models.go

```diff
@@ -103,7 +137,78 @@ type Estimator interface {
 	// attempts must:
 	//   - be sorted in order from highest price to lowest price
 	//   - all be of transaction type 0x2
-	BumpDynamicFee(ctx context.Context, original DynamicFee, gasLimit uint32, maxGasPriceWei *assets.Wei, attempts []PriorAttempt) (bumped DynamicFee, chainSpecificGasLimit uint32, err error)
+	BumpDynamicFee(ctx context.Context, original DynamicFee, gasLimit uint32, maxGasPriceWei *assets.Wei, attempts []EvmPriorAttempt) (bumped DynamicFee, chainSpecificGasLimit uint32, err error)
+}
+
+type EvmFee struct {
+	Legacy  *assets.Wei
+	Dynamic *DynamicFee
+}
+
+// FeeEstimator provides a generic interface for fee estimation
+//
+//go:generate mockery --quiet --name FeeEstimator --output ./mocks/ --case=underscore
+type FeeEstimator[HEAD any, FEE any, MAXPRICE any, HASH any] interface {
```

Comment:
Could you move all generic chain agnostic new interfaces that you add, to a separate common package?
This way, it is easy to review where the new interfaces are?

Btw, I am using the following directory for generic interfaces:
common/txmgr/types

Reply:
it might be better to leave it for now otherwise there might be some merge conflicts? once your PR gets merged with the new folder/module I can move it

I can mark a TODO to extract for now?

Reply:
Ok, up to you, but if you just move to your own new file name called Fees.go in same subdir as me, or even a different package for now, it should be ok.
It might make easier for reviewers to see what generic interfaces are there easily.

Reply:
moved this to a new module `chainlink/common/txmgr/types`